console/v1/types_console_yaml_sample.go

```diff
@@ -0,0 +1,26 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleYAMLSample is an extension for customizing OpenShift web console YAML samples.
+type ConsoleYAMLSample struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleYAMLSampleSpec `json:"spec"`
+}
+
+// ConsoleYAMLSampleSpec is the desired YAML sample configuration.
```

Comment:
How does this know what `resource` to reference in the console?  I'm assuming its a `customyamlsample` for `deployments`, or another resource.

Reply:
Here is the proposal for target resource in the console PR: https://github.com/openshift/console/pull/2889/files#r331471061

Reply:
so is that going to be added to this ConsoleYAMLSample api?

Reply:
> so is that going to be added to this ConsoleYAMLSample api?

Yes, we'll need that to know the kind of resource. The alternative is to try to parse the YAML sample content for all types to filter based on group/version/kind, but I'm not sure that's practical.

Reply:
well, the other option, if you foresee a need to reuse yaml samples for different resources, would be to have two different CRDs:


1) yaml sample
2) yaml sample reference/link to resource type to use it for

So you can create a single (1) and then multiple (2)s.

but i don't know if that is a valid use case or not.  Or even if it's valid, if it would be common enough to be worth the effort over just creating duplicate ConsoleYamlSample objects.

Reply:
Not really sure what would be the use-case for reusing the CR for different resources.
If the would be one of the solutions would be as @spadgett suggested in [comment](https://github.com/openshift/console/pull/2889/files#r331471061)  to have `TargetResource` that would be an array
```go
type ConsoleYAMLSampleSpec struct {
	TargetResource []metav1.TypeMeta `json:",targetResource"`
...
```

Thinking that if it wouldn't make more sense to have a multiple `YAML`s for a single `TargetResource` ?

Reply:
I'm not sure what the use case is for having more than one target resource since the API group / version / kind is also in the YAML. These should almost always match.

We do have one edge case in the UI where we show Roles and Cluster Roles in the same sidebar, but I think this pretty rare. There's always the workaround of creating more than one sample.